src/main/java/org/cactoos/scalar/AvgOf.java

```diff
@@ -250,16 +250,16 @@ public AvgOf(final Float... src) {
             }
             return sum / total;
         }, () -> {
-            float sum = 0.0f;
-            float total = 0.0f;
-            for (final float val : src) {
-                sum += val;
+            double sum = 0.0d;
+            double total = 1.0d;
+            for (final double val : src) {
+                sum += (val - sum) / total;
                 total += 1.0f;
             }
             if (total == 0.0f) {
                 total = 1.0f;
             }
```

Comment:
Aren't these IFs obsolete?

Reply:
@svendiedrichsen yes, you're right! Thanks.